### PR TITLE
refactor: Change AWSLambda example code

### PR DESCRIPTION
**Pull request by sdelamo**
This changes the sample code to use aws-lambda-events. Both app and function handle a AWS API Gateway proxy event.

### CODE REVIEW

File: starter-core/src/main/java/io/micronaut/starter/feature/function/awslambda/template/awsLambdaFunctionRequestHandlerGroovy.rocker.raw [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+package @project.getPackageName();
+}
+
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyRequestEvent;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyResponseEvent;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.micronaut.function.aws.MicronautRequestHandler;
+import jakarta.inject.Inject;
+import java.util.Collections;
```
Comment (timyates) [inline starter-core/src/main/java/io/micronaut/starter/feature/function/awslambda/template/awsLambdaFunctionRequestHandlerGroovy.rocker.raw]: Remove semi-colons (from package too)

And we don't need `import java.util.Collections;` as this is implicit

Comment (timyates) [reply starter-core/src/main/java/io/micronaut/starter/feature/function/awslambda/template/awsLambdaFunctionRequestHandlerGroovy.rocker.raw]: There are more semi-colons below 👇

File: starter-core/src/main/java/io/micronaut/starter/feature/function/awslambda/template/awsLambdaFunctionRequestHandlerGroovyJunit.rocker.raw [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        if (handler != null) {
+            handler.getApplicationContext().close()
+        }
+    }
+
+    @@Test
+    void testHandler() {
+        APIGatewayProxyRequestEvent request = new APIGatewayProxyRequestEvent()
+        request.setHttpMethod("GET")
+        request.setPath("/")
```
Comment (timyates) [inline starter-core/src/main/java/io/micronaut/starter/feature/function/awslambda/template/awsLambdaFunctionRequestHandlerGroovyJunit.rocker.raw]: nit: could be:

```
        request.httpMethod = "GET"
        request.path = "/"
```

File: starter-core/src/main/java/io/micronaut/starter/feature/function/awslambda/template/awsLambdaFunctionRequestHandlerSpock.rocker.raw [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+    @@AutoCleanup
+    @@Shared
+    FunctionRequestHandler handler = new FunctionRequestHandler()
+
+    void "test Handler"() {
+        given:
+        APIGatewayProxyRequestEvent request = new APIGatewayProxyRequestEvent()
+        request.setHttpMethod("GET")
+        request.setPath("/")
```
Comment (timyates) [inline starter-core/src/main/java/io/micronaut/starter/feature/function/awslambda/template/awsLambdaFunctionRequestHandlerSpock.rocker.raw:27]: nit: could be:

```
        request.httpMethod = "GET"
        request.path = "/"
```

File: starter-core/src/main/java/io/micronaut/starter/feature/function/awslambda/template/homeControllerGroovyJunit.rocker.raw [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+import com.amazonaws.serverless.proxy.model.AwsProxyRequest
+import com.amazonaws.serverless.proxy.model.AwsProxyResponse
+import com.amazonaws.services.lambda.runtime.Context
+import com.fasterxml.jackson.core.JsonProcessingException
+import com.fasterxml.jackson.databind.ObjectMapper
+import io.micronaut.function.aws.proxy.MicronautLambdaHandler
+import org.junit.jupiter.api.AfterAll
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Test
+import static org.junit.jupiter.api.Assertions.assertEquals;
```
Comment (timyates) [inline starter-core/src/main/java/io/micronaut/starter/feature/function/awslambda/template/homeControllerGroovyJunit.rocker.raw]: semi-colon

**Review by timyates — APPROVED** [review]
Some groovy style comments, but not enough to block it

I tried generating projects with

```
./gradlew :micronaut-cli:shaJar
```

And then

```
java -jar starter-cli/build/libs/micronaut-cli-3.5.0-SNAPSHOT-all.jar \
    create-function-app \
    --build=gradle \
    --jdk=11 \
    --lang=groovy \
    --test=spock \
    --features=aws-lambda-custom-runtime \
    temp_gr_sp
```

For `java`, `kotlin` and `groovy` with `junit` and `spock`

Then I imported them into IntelliJ and checked the tests ran

I didn't check Maven

Was there anything else?

File: starter-core/src/main/java/io/micronaut/starter/feature/function/awslambda/template/homeControllerGroovyJunit.rocker.raw (hunk 4 above)
Comment (sdelamo) [reply starter-core/src/main/java/io/micronaut/starter/feature/function/awslambda/template/homeControllerGroovyJunit.rocker.raw]: resolved via https://github.com/micronaut-projects/micronaut-starter/pull/1209/commits/851470e7ee8747c5aef117b3558762162ff73a96

File: starter-core/src/main/java/io/micronaut/starter/feature/function/awslambda/template/awsLambdaFunctionRequestHandlerSpock.rocker.raw (hunk 3 above)
Comment (sdelamo) [reply starter-core/src/main/java/io/micronaut/starter/feature/function/awslambda/template/awsLambdaFunctionRequestHandlerSpock.rocker.raw:27]: resolved via https://github.com/micronaut-projects/micronaut-starter/pull/1209/commits/851470e7ee8747c5aef117b3558762162ff73a96

File: starter-core/src/main/java/io/micronaut/starter/feature/function/awslambda/template/awsLambdaFunctionRequestHandlerGroovyJunit.rocker.raw (hunk 2 above)
Comment (sdelamo) [reply starter-core/src/main/java/io/micronaut/starter/feature/function/awslambda/template/awsLambdaFunctionRequestHandlerGroovyJunit.rocker.raw]: resolved via https://github.com/micronaut-projects/micronaut-starter/pull/1209/commits/851470e7ee8747c5aef117b3558762162ff73a96

File: starter-core/src/main/java/io/micronaut/starter/feature/function/awslambda/template/awsLambdaFunctionRequestHandlerGroovy.rocker.raw (hunk 1 above)
Comment (sdelamo) [reply starter-core/src/main/java/io/micronaut/starter/feature/function/awslambda/template/awsLambdaFunctionRequestHandlerGroovy.rocker.raw]: resolved via https://github.com/micronaut-projects/micronaut-starter/pull/1209/commits/851470e7ee8747c5aef117b3558762162ff73a96

**Pull request merged**